firebase-firestore/src/main/java/com/google/firebase/firestore/util/ExponentialBackoff.java

```diff
@@ -151,6 +161,9 @@ public void backoffAndRun(Runnable task) {
     } else if (currentBaseMs > maxDelayMs) {
       currentBaseMs = maxDelayMs;
     }
+
+    // Reset max delay to the default.
+    maxDelayMs = DEFAULT_BACKOFF_MAX_DELAY_MS;
```

Comment:
This ignored the comments I made on the last version of this PR.

The max delay was configured in the constructor and may not actually be `DEFAULT_BACKOFF_MAX_DELAY_MS`. You can't read this constant here.

Instead, save two max delay fields in the constructor. One should be final, as maxDelayMs is today and it should be used here as the value to which to reset. The second value should be the one you actually bounce around based on which kind of error is in effect.

Reply:
Done (attempted).